mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -917,11 +916,15 @@ private[spark] object RandomForest extends Logging {
       // being spun up that will definitely do no work.
       val numPartitions = math.min(continuousFeatures.length, input.partitions.length)
 
+      val numInput = input.count()
```

Comment:
we can get this from the `metadata.numExamples * fraction` operation in the calling method in order to avoid another job to perform the count